File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res = 0
        sell_max = 0
        for price in reversed(prices):
            if sell_max - price > 0:
                res = max(res, sell_max - price)
            sell_max = max(sell_max,price)

        return res
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int sell_max = 0;
        for (int i=prices.size()-1; i>-1; i--) {
            if (sell_max - prices[i]>0) {
                res = max(res, sell_max-prices[i]);
            }
            sell_max = max(sell_max, prices[i]);
        }
        return res;
    }
};